src/worker.cpp

```diff
@@ -1185,4 +1187,251 @@ void Worker::collectFrusFromJson()
+    {
+        fillVPDMap(l_source2Path, l_source2VpdMap);
+    }
+
+    // Implement backup and restore for IPZ type VPD
+    if (auto l_backupMapPtr = std::get_if<types::IpzBackupMap>(&l_backupMap))
+    {
+        types::IPZVpdMap* l_source1VpdPtr = nullptr;
+        if (!(l_source1VpdPtr =
+                  std::get_if<types::IPZVpdMap>(&io_source1VpdMap)))
+        {
+            if (std::holds_alternative<std::monostate>(io_source1VpdMap))
+            {
+                l_source1VpdPtr = new types::IPZVpdMap();
+            }
+            else
+            {
+                logging::logMessage("Source1 VPD is not of IPZ type.");
+                return;
+            }
+        }
+
+        types::IPZVpdMap* l_source2VpdPtr;
+        if (!(l_source2VpdPtr =
+                  std::get_if<types::IPZVpdMap>(&l_source2VpdMap)))
+        {
+            if (std::holds_alternative<std::monostate>(l_source2VpdMap))
+                l_source2VpdPtr = new types::IPZVpdMap();
+            else
+            {
+                logging::logMessage("Source2 VPD is not of IPZ type.");
+                return;
+            }
+        }
+
+        backupAndRestoreIpzVpd(*l_backupMapPtr, *l_source1VpdPtr,
+                               *l_source2VpdPtr);
+    }
+    // Note: add implementation here to support any other VPD type.
+}
+
+void Worker::backupAndRestoreIpzVpd(const types::IpzBackupMap& i_backupInfoMap,
+                                    types::IPZVpdMap& io_source1VpdMap,
+                                    const types::IPZVpdMap& i_source2VpdMap)
+{
+    std::string l_source1Location;
+    std::string l_source1Path;
+    getBackupAndRestorePathInfo("Source1", l_source1Location, l_source1Path);
+
+    std::string l_source2Location;
+    std::string l_source2Path;
+    getBackupAndRestorePathInfo("Source2", l_source2Location, l_source2Path);
+
+    for (const auto& l_recordKwInfo : i_backupInfoMap)
+    {
+        const std::string& l_recordName = l_recordKwInfo.first;
+
+        if (!io_source1VpdMap.empty() &&
+            io_source1VpdMap.find(l_recordName) == io_source1VpdMap.end())
+        {
+            logging::logMessage(
+                "Record: " + l_recordName +
+                ", is not found in the source1 path: " + l_source1Path);
+            continue;
+        }
+
+        if (!i_source2VpdMap.empty() &&
+            i_source2VpdMap.find(l_recordName) == i_source2VpdMap.end())
+        {
+            logging::logMessage(
+                "Record: " + l_recordName +
+                ", is not found in the source2 path: " + l_source2Path);
+            continue;
+        }
+
+        for (const auto& l_keywordInfo : l_recordKwInfo.second)
+        {
+            const std::string& l_keyword = get<0>(l_keywordInfo);
+            const auto& l_defaultBinaryValue = get<1>(l_keywordInfo);
+            bool l_isPelRequired = get<2>(l_keywordInfo);
+
+            const auto& l_source1BinaryValue = getIpzKeywordValue(
```

Comment:
The checks that are being done in this API "getIpzKeywordValue" is already being done before calling this API.
Why do we need to repeat map.find operation? Can the existing one be re-used?

Reply:
> We do require parameter validation as required in any other API's.

Correct but we can avoid a map find (which can be an expensive operation for large maps) but just pulling that implementation here.
Anyways the API is not giving a generic usability, it is very specific to back up and restore.

Reply:
Removed this api, instead having implementation here itself.